### ISSUE TITLE
Pretty sure patch in Pull Request #1313 is incorrect

### ISSUE THREAD

**Issue by jasonrohrer** [huggingface/transformers#2180]
## 🐛 Bug

The bug was introduced in pull request #1313

If a stop_token is specified, but does not exist in **text**, then the last character of the text string is trimmed off.  text.find will return -1 in that case, which seems to remove the last character from the string.

Example:
--stop_token="wake"

`text = tokenizer.decode(o, clean_up_tokenization_spaces=True)`

text = ' fishing with two ladies and the same two ladies had been fishing with him and all three ladies had run'

` text = text[: text.find(args.stop_token) if args.stop_token else None]`

text = ' fishing with two ladies and the same two ladies had been fishing with him and all three ladies had ru'

See how that final 'n' is trimmed off, even though "wake" does not occur in this string?

The fix should be to first find out if stop_token occurs in the text before doing the trimming, maybe:

```
if args.stop_token :
    loc = text.find(args.stop_token)
    if loc != -1 :+1:
        text = text[:loc]
```

**Comment by jasonrohrer** [comment]
And yes, the above code does fix it.

Example:
--stop_token="wake"

text = ' to the same conclusion: that her husband, the guy she had chosen for her favorite, had been'

trimmed text = ' to the same conclusion: that her husband, the guy she had chosen for her favorite, had been'



text = ' murdered the previous spring.  Oedipa closed her eyes and tried to wake up.'

trimmed text = ' murdered the previous spring.  Oedipa closed her eyes and tried to '

**Comment by LysandreJik** [comment]
Thank you for raising this issue @jasonrohrer, there indeed was an error with the stop token. It's been fixed in 18a879f.
